packages/angular_devkit/build_angular/src/protractor/trusted_types_spec.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Architect } from '@angular-devkit/architect';
+import { normalize, virtualFs } from '@angular-devkit/core';
+import { createArchitect, host, protractorTargetSpec } from '../test-utils';
+
+describe('Trusted Types enforcement', () => {
+  let architect: Architect;
+
+  beforeEach(async () => {
+    await host.initialize().toPromise();
+    architect = (await createArchitect(host.root())).architect;
+  });
+
+  afterEach(() => host.restore().toPromise());
+
+  it('works in a basic application using JIT', async () => {
+    const overrides = {
+      headers: {
+        'Content-Security-Policy':
+          "trusted-types angular angular#unsafe-bypass angular#unsafe-jit; require-trusted-types-for 'script';",
+      },
+    };
+
+    const run = await architect.scheduleTarget(protractorTargetSpec, overrides);
+
+    await expectAsync(run.result).toBeResolvedTo(jasmine.objectContaining({ success: true }));
+
+    await run.stop();
+  });
+
+  it('works in a basic application using AOT', async () => {
+    const overrides = {
+      headers: {
+        'Content-Security-Policy':
+          "trusted-types angular angular#unsafe-bypass; require-trusted-types-for 'script';",
+      },
+      devServerTarget: 'app:serve:production',
```

Comment:
I'm not super familiar with this testing setup so it's possible that I'm wrong, but this doesn't look right to me.

The AOT mode is on by default now, and even if it wasn't the case in this test setup (see my comment above), we'd still want to selectively turn on the AOT and use the dev mode pipeline, so that we are testing the workflow that developers would use most often.

btw we could reduce the number of tests, by using lazy imports by default. It's not strictly necessary, but it would reduce the number of combinations we need to test.

Let's double check how the setup works for these tests and ensure that we are testing these four combinations:
- JIT (devserver) + lazy imports
- AOT (devserver) + lazy imports
- JIT (prodserver) + lazy imports
- AOT (prodserver) + lazy imports